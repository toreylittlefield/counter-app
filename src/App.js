  // src/App.js

  import React, { useState } from 'react';

  const App = () => {
    const [count, setCount] = useState(0);

    const makeIncrementer = amount => () =>
        setCount(prevState => (prevState + amount));

    const increment = makeIncrementer(1);

    const decrement = makeIncrementer(-1);

      return (
        <div>
          <p>Count: {count}</p>
          <button className="increment" onClick={increment}>Increment count</button>
          <button className="decrement" onClick={decrement}>Decrement count</button>
        </div>
      )
  }

  export default App;