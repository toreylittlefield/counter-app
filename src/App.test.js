import { render, screen } from '@testing-library/react';
import ReactDOM from 'react-dom';
import App from './App';

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Addition', () => {
  it('knows that 2 and 2 make 4', () => {
    expect(2 + 2).toBe(5);
  });
});