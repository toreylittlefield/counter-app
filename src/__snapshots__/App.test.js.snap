// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div>
  <p>
    Count: 
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
  >
    Increment count
  </button>
  <button
    className="decrement"
    onClick={[Function]}
  >
    Decrement count
  </button>
</div>
`;
